pkg/model/tree.go

```diff
@@ -0,0 +1,471 @@
+package model
+
+import (
+	"container/heap"
+	"fmt"
+	"io"
+	"sort"
+	"sync"
+
+	dvarint "github.com/dennwc/varint"
+	"github.com/pyroscope-io/pyroscope/pkg/util/varint"
+	"github.com/xlab/treeprint"
+)
+
+type Tree struct {
+	root []*node
+}
+
+func emptyTree() *Tree {
+	return &Tree{}
+}
+
+func newTree(stacks []stacktraces) *Tree {
```

Comment:
is this for backward compatibility ?

Reply:
Used in tests. Refactored helpers to `_test.go`